File: backend/Reimbursement-Calculation-App/src/main/java/pl/Alski/entity/claim/DTO/ClaimRequest.java
package pl.Alski.entity.claim.DTO;

import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
public class ClaimRequest {
    private LocalDate startDate;
    private LocalDate endDate;
    private List<ReceiptDTO> receipts;
    private List<LocalDate> reimbursedDaysWithAllowance;
    private int distanceInKMForCarMileage;
    private String carPlates;
    private double totalReimbursementAmount;
    private int userId;
}

File: frontend/ReimbursementApp/src/app/common/claim-request.ts
import { Receipt } from "./receipt";

export class ClaimRequest {
    startDate: Date = new Date();
  endDate: Date = new Date();
  receipts: Receipt[] = [];
  reimbursedDaysWithAllowance: Date[] = [];
  distanceInKMForCarMileage: number = 0;
  carPlates: string = '';
  totalReimbursementAmount: number = 0;
  userId: number = 0;
}
